src/components/global/Header/ConnectWalletButton.tsx

```diff
@@ -118,40 +143,139 @@ const WalletButton = () => {
+              }}
+            >
+              <img
+                src="/images/icon-copy.svg"
+                height="15px"
+                width="16px"
+                alt="Copy Address"
+              />
+              {copiedAddress ? `Copied!` : `Copy Address`}
+            </TokenModal.WalletAction>
+            <NextLink
+              href={`https://etherscan.io/address/${walletWeb3ReactContext.account}`}
+            >
+              <TokenModal.WalletAction>
+                <img
+                  src="/images/icon-export.svg"
+                  height="15px"
+                  width="15px"
+                  alt="Open in Etherscan"
+                />
+                Etherscan
+              </TokenModal.WalletAction>
+            </NextLink>
+            <TokenModal.WalletAction
+              onClick={() => walletWeb3ReactContext.deactivate()}
+            >
+              <img
+                src="/images/icon-wallet.svg"
+                height="15px"
+                width="15px"
+                alt="Change Wallet"
+              />
+              Change Wallet
+            </TokenModal.WalletAction>
+          </TokenModal.WalletActionsRow>
+          <TokenModal.BigRow>
+            <TokenModal.BankTitle>Bankless DAO / $BANK</TokenModal.BankTitle>
+            <TokenModal.BigBank>
+              <strong>{whaleBalance}</strong>
+              <img
+                onMouseEnter={() => {
+                  setAnimateToken(true)
+                }}
+                onMouseLeave={() => {
+                  setAnimateToken(false)
+                }}
+                src={
+                  animateToken
+                    ? '/images/token-3d-animated.gif'
+                    : '/images/token-3d.png'
+                }
+                height="85px"
+                width="85px"
+                alt="3d token icon"
+              />
+            </TokenModal.BigBank>
+            <TokenModal.BankBalance>
+              {commaBalance} <strong>BANK</strong>
+            </TokenModal.BankBalance>
+            {unclaimedTokens ? (
+              <TokenModal.UnclaimedNotice
+                onClick={() => {
+                  setModalOpen(false)
+                  router.push('/claim')
+                }}
+              >
+                <img
+                  src="/images/icon-warning.svg"
+                  height="15px"
+                  width="15px"
+                  alt="warning"
+                />
+                <span>Unclaimed Tokens Now Available</span>
+                <img
+                  src="/images/icon-warning.svg"
+                  height="15px"
+                  width="15px"
+                  alt="warning"
+                />
+              </TokenModal.UnclaimedNotice>
+            ) : null}
+          </TokenModal.BigRow>
+          <TokenModal.ToolRow>
+            <TokenModal.ToolItem onClick={() => addToMetaMask()}>
+              Add BANK to MetaMask
```

Comment:
is there anything in the provider that would tell us we're connected to MM. Would love to hide this for WalletConnect

Reply:
super bonus if it can detect if BANK is already added

Reply:
Great idea, this is definitely possible

Reply:
> super bonus if it can detect if BANK is already added

Checked into this, and detecting tracked assets is [not possible with the current Metamask](https://stackoverflow.com/questions/67771978/is-it-possible-to-read-fetch-added-customised-token-in-metamask-in-react)

Reply:
> is there anything in the provider that would tell us we're connected to MM. Would love to hide this for WalletConnect

This is now hidden when using Wallet Connect in the latest commits below.